extensions/sql-migration/src/api/azure.ts

```diff
@@ -39,22 +39,78 @@ export type SqlManagedInstance = AzureProduct;
 export async function getAvailableManagedInstanceProducts(account: azdata.Account, subscription: Subscription): Promise<SqlManagedInstance[]> {
 	const api = await getAzureCoreAPI();
 
-	const result = await api.runGraphQuery<SqlManagedInstance>(account, [subscription], false, `where type == "${azureResource.AzureResourceType.sqlManagedInstance}"`);
+	const result = await api.getSqlManagedInstances(account, [subscription], false);
 	return result.resources;
 }
 
 export type SqlServer = AzureProduct;
 export async function getAvailableSqlServers(account: azdata.Account, subscription: Subscription): Promise<SqlServer[]> {
 	const api = await getAzureCoreAPI();
 
-	const result = await api.runGraphQuery<SqlServer>(account, [subscription], false, `where type == "${azureResource.AzureResourceType.sqlServer}"`);
+	const result = await api.getSqlServers(account, [subscription], false);
 	return result.resources;
 }
 
 export type SqlVMServer = AzureProduct;
 export async function getAvailableSqlVMs(account: azdata.Account, subscription: Subscription): Promise<SqlVMServer[]> {
 	const api = await getAzureCoreAPI();
 
-	const result = await api.runGraphQuery<SqlVMServer>(account, [subscription], false, `where type == "${azureResource.AzureResourceType.virtualMachines}" and properties.storageProfile.imageReference.publisher == "microsoftsqlserver"`);
+	const result = await api.getSqlVMServer(account, [subscription], false);
 	return result.resources;
 }
+
+export type StorageAccount = AzureProduct;
+export async function getAvailableStorageAccounts(account: azdata.Account, subscription: Subscription): Promise<StorageAccount[]> {
+	const api = await getAzureCoreAPI();
+	const result = await api.getStorageAccounts(account, [subscription], false);
+	sortResourceArrayByName(result.resources);
+	return result.resources;
+}
+
+export type FileShares = AzureProduct;
+export async function getFileShares(account: azdata.Account, subscription: Subscription, storageAccount: StorageAccount): Promise<FileShares[]> {
+	const api = await getAzureCoreAPI();
+	const url = `https://management.azure.com` +
+		`/subscriptions/${subscription.id}` +
+		`/resourceGroups/${storageAccount.resourceGroup}` +
+		`/providers/Microsoft.Storage/storageAccounts/${storageAccount.name}` +
+		`/fileServices/default/shares?api-version=2019-06-01`;
+
+	let result = await api.makeHttpGetRequest(account, subscription, true, url);
```

Comment:
So why are you using the HTTP call for this anyways? Does the ARM API not support getting file shares?